editor/gui/scene_tree_editor.cpp

```diff
@@ -924,12 +924,19 @@ void SceneTreeEditor::_update_tree(bool p_scroll_to_selected) {
 		// If pinned state changed, update the currently pinned node.
 		if (AnimationPlayerEditor::get_singleton()->is_pinned() != node_cache.current_has_pin) {
 			node_cache.current_has_pin = AnimationPlayerEditor::get_singleton()->is_pinned();
-			node_cache.mark_dirty(pinned_node);
+			if (node_cache.has(pinned_node)) {
+				node_cache.mark_dirty(pinned_node);
+			}
 		}
 		// If the current pinned node changed update both the old and new node.
 		if (node_cache.current_pinned_node != pinned_node) {
-			node_cache.mark_dirty(pinned_node);
-			node_cache.mark_dirty(node_cache.current_pinned_node);
+			// get_editing_node() will return deleted nodes. If the nodes are not in cache don't try to mark them.
```

Comment:
Why not also fix the upstream issue and null `original_node` when `player` is cleared?

Reply:
Apologies I missed this comment before merging, otherwise I'd have waited for @hpvb to reply first.

Reply:
No worries, ultimately the current code only uses `get_editing_node` in two places and both would now be safe handling that pointer but it seems like it would be better to prevent that from being returned in the first place.

Reply:
Because this was a release blocker I decided to do a local fix. The current way the editor pinning works is kind of broken, editors can be pinned but the status of whether or not it is pinned is stored in the button toggle state, which gets reset when switching scenes. 

The behavior is kind of silly as it is now, and was silly before any of my changes. I'm planning to do a ui fix pr separately to fix the pinning behavior in general.